### ISSUE TITLE
Wrong Locality association at a street - Sarentino

### ISSUE THREAD

**Issue by moberrauch** [osmandapp/OsmAnd#20045]
### Description

I found a street, where the "calculated address", which you get by long-pressing on the map, has indicates a wrong municipality. The data in OSM is correct, so this is a presentation / association issue by OsmAnd.

### Steps to reproduce

Go to 46.64266° N, 11.35479° E and long-tap along the street "Andreas-Hofer". You get "Andreas-Hofer-Straße - Via Andreas Hofer, Gargazon - Gargazzone", which indicates the street being in the municipality of "Gargazon" (a village a few kilometers south-west), but actually it should be in "Sarnthein" ("Sarentino" in italian).
It's correct for nearby streets like "Runggener-Straße" or "Griesplatz".

### Actual result

Andreas-Hofer-Straße - Via Andreas Hofer, Gargazon - Gargazzone

### Expected result

Andreas-Hofer-Straße - Via Andreas Hofer, Sarnthein - Sarentino

### Your Environment (required)

**WARNING** Crash-Logs **MAY** contain information you deem sensitive.
Review this **CAREFULLY** before posting your issue!

```
OsmAnd Version: 4.7.10
Android/iOS version: Android 14 (LineageOS 21)
Device model: Fairphone 4
Crash-Logs: n.a.
```

**Comment by yuriiurshuliak** [comment]
The bug has been reproduced. To reproduce it, navigate to the specified coordinates and click anywhere along "Andreas-Hofer" street. You will notice it is incorrectly marked as being in the "Gargazon" municipality instead of "Sarnthein". Here is an example on OSM: https://www.openstreetmap.org/node/2755958093

OsmAnd~ 4.8.0#2469m, released: 2024-06-10

<table style="width:100%">
  <tr>
<th><img src="https://github.com/osmandapp/OsmAnd/assets/127092082/9d453296-dd62-488d-ad3c-35478c2467d8"  width="300" height="auto"/> </th>
  </tr>
</table>

**Comment by vshcherb** [comment]
Point - https://nominatim.openstreetmap.org/ui/details.html?osmtype=N&osmid=2755958093&class=place
Polygon - https://nominatim.openstreetmap.org/ui/details.html?osmtype=R&osmid=47282&class=boundary
Wrong polygon https://nominatim.openstreetmap.org/ui/details.html?osmtype=R&osmid=47260&class=boundary

**Comment by ivanPyrohivskyi** [comment]
Problem with lat lon searching https://github.com/osmandapp/OsmAnd/pull/21226

**Comment by Zirochkabila** [comment]
iOS - still present
`46.64238° N, 11.35301° E`

<table style="width:100%">
  <tr>
    <th>OsmAnd~ 5.0.0#3875m, Released: 2024-12-04</th>
    <th>4.9.0.11</th>
  </tr>
  <tr>
<th><img src="https://github.com/user-attachments/assets/1e721113-2983-4d91-b13a-f533aaafe786"  width="450" height="auto"/> </th>
 <th><img src="https://github.com/user-attachments/assets/45dd475d-9828-42bf-9380-24d0e6e604a1"  width="300" height="auto"/> </th>
  </tr>
</table>

**Comment by ivanPyrohivskyi** [comment]
[Changes](https://github.com/osmandapp/OsmAnd-core/pull/800) is not included in the release iOS version.

**Comment by ivanPyrohivskyi** [comment]
Changes applied in 4.9

**Comment by ivanPyrohivskyi** [comment]
Changes applied in r4.9
https://github.com/osmandapp/OsmAnd-core/compare/c2ec58f82a789ab3f4b1d9a75117047983ab58d5...3a6982ec9dd8b0ee882d33829b54d171ded37981
<img src="https://github.com/user-attachments/assets/5628c993-3bd7-49ed-82e3-37e10de57942" width="400"/>

**Comment by vshcherb** [comment]
Yes, not clear what's been fixed as Gargazzone is far away but it has boundary, Sarentino doesn't have boundary but it's much closer.

![Image](https://github.com/user-attachments/assets/c5f1f87e-e9a5-4b1d-a133-a5017ff0e441)

**Comment by Zirochkabila** [comment]
# Testing 3
`OsmAnd~ 5.0.0#4474m, Released: 2025-02-26`
`4982`


1. iOS - city name is not displayed
[OSM](https://www.openstreetmap.org/node/8953392959#map=19/46.649533/11.359088)


<details><summary>iOS</summary>
<table style="width:100%">
  <tr>
<th><img src="https://github.com/user-attachments/assets/58c6f8ec-54b7-4e3b-a42c-9e2d0476c203"  width="300" height="auto"/> </th>
 <th><img src="https://github.com/user-attachments/assets/6938ff8b-5f28-4f36-a5a0-260fe2af3cd6"  width="300" height="auto"/> </th>
  </tr>
</table>  
</details> 

________

2. iOS - wrong city name
[OSM](https://www.openstreetmap.org/node/2755960196#map=19/46.650374/11.357827)

<details><summary>iOS</summary>
<table style="width:100%">
  <tr>
<th><img src="https://github.com/user-attachments/assets/8e588214-0486-46f5-96d1-83d9e439c205"  width="300" height="auto"/> </th>
 <th><img src="https://github.com/user-attachments/assets/829dfc3c-b53f-441c-a206-e6b18f8023e2"  width="300" height="auto"/> </th>
  </tr>
</table>  
</details> 

_______

3. iOS - address is missing
[OSM](https://www.openstreetmap.org/node/2755955807#map=19/46.649697/11.357954)


<details><summary>iOS</summary>
<table style="width:100%">
  <tr>
<th><img src="https://github.com/user-attachments/assets/8fd388cb-d31b-401f-9374-2c9863a0a6b2"  width="300" height="auto"/> </th>
 <th><img src="https://github.com/user-attachments/assets/412915c8-5561-4d12-a306-b6e303ee5381"  width="300" height="auto"/> </th>
  </tr>
</table>  
</details>

**Comment by dmpr0** [comment]
Both issues mentioned in https://github.com/osmandapp/OsmAnd/issues/20045#issuecomment-2634374860 is fixed.

<details>
    <summary>Screenshot</summary>
    <table style="width:100%">
      <tr>
        <th></th>
        <th></th>
      </tr>
      <tr>
        <td><img src="https://github.com/user-attachments/assets/55db78ca-599c-412d-8167-e9e556f39c05"  width="300" height="auto"/> </td>
        <td><img src="https://github.com/user-attachments/assets/8a41f561-9008-480a-88b5-67fcd768b10e"  width="300" height="auto"/> </td>
      </tr>
    </table>   
</details>
